doc/monitoring/index.rst

```diff
@@ -64,6 +64,32 @@ in:
 
 It is good to outline that the main tool allowing users to observe and monitor a Ceph cluster is the **Ceph dashboard**. It provides graphics where the most important cluster and service metrics are represented. Most of the examples in this document are extracted from the dashboard graphics or extrapolated from the metrics exposed by the Ceph dashboard.
 
+Ceph daemon health metrics
+==========================
+
+The Ceph exporter provides a metric called ``ceph_daemon_socket_up`` that reports the health status of all Ceph daemons exposing an admin socket. These include the main Ceph daemons, encompassing any daemon capable of exposing performance counters.
```

Comment:
I understand the idea of using the socket status to see if the daemon is OK or not, but I'd recommend having a specific health field. A daemon can be reachable (network is OK) but not functioning because of XYZ (that would depend on each case). I'd recommend separating the health status of the daemon from its ability to communicate and have a separate metrics for both of them. For example in k8s world you have the **Liveness Probe** to see if the daemon is reachable or not and the **Readiness Probe** to see if the pod is ready do what it has to do.

Reply:
> Maybe it was not a good idea to use the caption "ceph daemon health metrics". In fact I have named the new metric "ceph_daemon_socket_up" ... which i think clearly indicates what is used for.
> 
> The thing is that if you have this metric returned with a value of 0, that clearly indicates that the daemon is not working properly, and therefore is not healthy (that is why used the "title ceph daemon health metrics", but we can change it for "Ceph daemon readiness metrics". In any case, how do you suggest to use as name for the new metric, and what is your specific proposal about the "titles" or explanation in the documentation, should we talk about readiness/liveness?
> 
> This modification, comes from the need to have something to indicate that an RBD mirror daemon was not working. ) [rhbz#2146728](https://bugzilla.redhat.com/show_bug.cgi?id=2146728)): In fact, i think this metric, is the best way to know that something wrong has happened with the daemon, due to the fact that the Ceph exporter communicates directly with the daemon through the admin socket, and it is not affected by any network issue. With this metric you can detect very easily a problem in the daemon, when the issue happened and if still is happening. Without this metric, to know abut the daemon health/readiness/liveness you need to use a combination of queries using "Ceph health" plus different set of metrics depending on the daemon in order to obtain the same information ( when the daemon stop responding and during how long.)
> 
> There is border use case, which is that the daemon does not answer request through admin socket, but still it answers using other communication ways.. (http/https for example). But even in this case, to have the metric "ceph_daemon_socket_up" reporting the problem is useful, because that means that we are not going to have any "perf counter dump" from the daemon, leaving blindly the daemon monitoring.

I see, thank you @jmolmo for the detailed explanation. Now I understand better what this metric is and what's the purpose of its usage. I'd say it seems more related to readiness (as if the daemon admin socket is not working, we can't send any admin cmds) that liveness (as the daemon could still be working and processing requests). The name you choose for the metric `ceph_daemon_socket_up` I think is clear and describes what it's reporting (not sure if it would be more clear by adding _admin so we end up with `ceph_daemon_admin_socket_up` but that's a subtle detail).